config/op-consumer/public/delta-context-config.js

```diff
@@ -0,0 +1,193 @@
+PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
+PREFIX ch: <http://data.lblod.info/vocabularies/contacthub/>
+PREFIX code: <http://lblod.data.gift/vocabularies/organisatie/>
+PREFIX dbpedia: <http://dbpedia.org/ontology/>
+PREFIX dc_terms: <http://purl.org/dc/terms/>
+PREFIX ere: <http://data.lblod.info/vocabularies/erediensten/>
+PREFIX euro: <http://data.europa.eu/m8g/>
+PREFIX euvoc: <http://publications.europa.eu/ontology/euvoc#>
+PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+PREFIX generiek: <https://data.vlaanderen.be/ns/generiek#>
+PREFIX lblodlg: <https://data.lblod.info/vocabularies/leidinggevenden/>
+PREFIX locn: <http://www.w3.org/ns/locn#>
+PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+PREFIX org: <http://www.w3.org/ns/org#>
+PREFIX organisatie: <https://data.vlaanderen.be/ns/organisatie#>
+PREFIX person: <http://www.w3.org/ns/person#>
+PREFIX persoon: <https://data.vlaanderen.be/ns/persoon#>
+PREFIX prov: <http://www.w3.org/ns/prov#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX regorg: <http://www.w3.org/ns/regorg#>
+PREFIX schema: <http://schema.org/>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>`
+
+
+const contextConfig = {
+  addTypes: {
+    scope: 'all', // 'inserts, 'deletes', 'all' or none. To add rdf:type to subjects of inserts, deletes or both
+    exhausitive: false, // true or false: find all types for a subject, even if one is already present in delta
+  },
+  contextQueries: [
+    {
+      trigger: { // subjectType or predicateValue
+        subjectType: "adms:Identifier"
+      },
+      queryTemplate: (subject) => `
+        ${PREFIXES}
+        CONSTRUCT {
+          ?identifier
+            a adms:Identifier;
+            generiek:gestructureerdeIdentificator ?gestructureerdeIdentificator;
+            skos:notation ?identificatorType.
+          ?gestructureerdeIdentificator
+            a generiek:GestructureerdeIdentificator;
+            generiek:lokaleIdentificator ?lokaleIdentificator.
+          ?bestuurseenheid
+            adms:identifier ?identifier.
+        } WHERE {
+          GRAPH <http://mu.semte.ch/graphs/ingest/op-public> {
+            VALUES ?identifier { ${subject} }
+            ?identifier
+              generiek:gestructureerdeIdentificator ?gestructureerdeIdentificator;
+              skos:notation ?identificatorType.
+            ?gestructureerdeIdentificator
+              generiek:lokaleIdentificator ?lokaleIdentificator.
+            ?bestuurseenheid
+              adms:identifier ?identifier.
+          }
+        }`
+    },
+    {
+      trigger: { // subjectType or predicateValue
+        predicateValue: "generiek:lokaleIdentificator"
+      },
+      queryTemplate: (subject) => `
+        ${PREFIXES}
```

Comment:
Still have to wrap my mind about it, but it blurs a bit the line between what the reasoner does, and what the context does. Implicitly, here, the context decides what extra triples can be deleted/inserted, whereas it feels more here it's a reasoner to do it.

Reply:
And i am not sure  EYE N3 Reasoner  supports this kind of expressivity

Reply:
The query here is only executed to (try) to add context to the deltas if the trigger matches.
The reasoner later does the mapping and filtering based on the configured rules and queries.

The reasoner does this in this scenario with deltas+context. Or just with the whole ingest graph during the initial sync (where this config is irrelevant).

So, not sure what you mean with the expressivity. If it's fetching additional data as input at runtime, then indeed it doesn't. Even though there technically is a built-in `log:semantics` which can fetch data - I think it's probably not most suitable option.

Let me know though if I misinterpreted your comment.